core/src/main/scala/org/apache/spark/storage/BlockObjectWriter.scala

```diff
@@ -123,12 +130,34 @@ private[spark] class DiskBlockObjectWriter(
    */
   private var numRecordsWritten = 0
 
+  private var sparkConf:SparkConf = null
+
   override def open(): BlockObjectWriter = {
     if (hasBeenClosed) {
       throw new IllegalStateException("Writer already closed. Cannot be reopened.")
     }
     fos = new FileOutputStream(file, true)
-    ts = new TimeTrackingOutputStream(fos)
+    var isEncryptedShuffle: Boolean = if (sparkConf != null) {
+      sparkConf.getBoolean("spark.encrypted.shuffle", false)
+    }
+    else {
+      false
+    }
+    if (isEncryptedShuffle) {
```

Comment:
In fact you could even inline the `conf.getBoolean` call here.

(BTW you have this code repeated later in the file, same comments apply there.)